src/librustc/session/config.rs

```diff
@@ -695,6 +722,17 @@ pub fn default_configuration(sess: &Session) -> ast::CrateConfig {
     if sess.opts.debug_assertions {
         ret.push(attr::mk_word_item(InternedString::new("debug_assertions")));
     }
+
+    sess.opts.cg.sanitize.map(|s| {
+        let name = match s {
+            Sanitize::Address => "sanitize_address",
+            Sanitize::Leak    => "sanitize_leak",
+            Sanitize::Memory  => "sanitize_memory",
+            Sanitize::Thread  => "sanitize_thread"
+        };
+        ret.push(attr::mk_word_item(InternedString::new(name)));
+    });
```

Comment:
Could you describe some sample scenarios you see for these? It looks like it's basically allowing you to do something like:

```rust
#[cfg(sanitize_address)]
fn foo() {}

#[cfg(not(sanitize_address))]
fn foo() {}
```

right?

In terms of bikeshedding, this may want to instead be:

```rust
#[cfg(sanitize = "address")]
```

(as that's what the CLI flag is)